protocol/bc/issuance_witness.go

```diff
@@ -1,9 +1,78 @@
 package bc
 
+import (
+	"fmt"
+	"io"
+
+	"chain/crypto/sha3pool"
+	"chain/encoding/blockchain"
+	"chain/errors"
+)
+
 type IssuanceWitness struct {
 	InitialBlock    Hash
 	AssetDefinition []byte
 	VMVersion       uint64
 	IssuanceProgram []byte
 	Arguments       [][]byte
 }
+
+func (iw *IssuanceWitness) AssetID() AssetID {
+	return ComputeAssetID(iw.IssuanceProgram, iw.InitialBlock, iw.VMVersion, iw.AssetDefinitionHash())
+}
+
+func (iw *IssuanceWitness) AssetDefinitionHash() (defhash Hash) {
+	if len(iw.AssetDefinition) == 0 {
```

Comment:
Hmm, I know we said we'd try to just merge this without teasing it apart more, but do you think it would be easy to pull those fixes out of this PR, and make this PR purely refactoring?